code/datums/abilities/flock/flockmind.dm

```diff
@@ -204,27 +204,35 @@
 
 /datum/targetable/flockmindAbility/healDrone
 	name = "Concentrated Repair Burst"
-	desc = "Fully heal a drone through acceleration of its repair processes."
+	desc = "Heal all flock units in an area."
 	icon_state = "heal_drone"
-	cooldown = 20 SECONDS
+	cooldown = 30 SECONDS
+	var/max_targets = 4 //maximum number of drones healed
 
-/datum/targetable/flockmindAbility/healDrone/cast(mob/living/critter/flock/drone/target)
+/datum/targetable/flockmindAbility/healDrone/cast(atom/target)
 	if(..())
 		return TRUE
-	if(!istype(target))
-		return TRUE
-	if (target.get_health_percentage() >= 1)
-		boutput(holder.get_controlling_mob(), "<span class='notice'>[target.real_name] has no damage!</span>")
-		return TRUE
-	if (isdead(target))
-		boutput(holder.get_controlling_mob(), "<span class='notice'>[target.real_name] is dead!</span>")
-		return TRUE
+	var/healed = 0
+	for (var/mob/living/critter/flock/flockcritter in range(3, target))
+		var/health_ratio = flockcritter.get_health_percentage()
+		if (isdead(flockcritter) || health_ratio >= 1)
+			continue
+		flockcritter.HealDamage("All", 30, 30) //half of a flockdrone's health
+		var/particles/healing/flock/particles = new
+		particles.spawning = 1 - health_ratio //more heal = more particles
+		flockcritter.UpdateParticles(particles, "flockmind_heal")
+		SPAWN(1.5 SECONDS)
+			particles.spawning = 0
+			sleep(1.5 SECONDS)
+			flockcritter.ClearSpecificParticles("flockmind_heal")
+		if (istype(flockcritter, /mob/living/critter/flock/drone))
+			healed++
+		if (healed > src.max_targets)
+			return
 
 	playsound(holder.get_controlling_mob(), 'sound/misc/flockmind/flockmind_cast.ogg', 80, 1)
-	boutput(holder.get_controlling_mob(), "<span class='notice'>You focus the flock's efforts on fixing [target.real_name]</span>")
-	target.HealDamage("All", 200, 200)
-	logTheThing(LOG_COMBAT, holder.get_controlling_mob(), "casts repair burst on [constructTarget(target)] at [log_loc(src.holder.owner)].")
-	target.visible_message("<span class='notice'><b>[target]</b> suddenly reforms its broken parts into a solid whole!</span>", "<span class='notice'>The flockmind has restored you to full health!</span>")
```

Comment:
Can see why it was left out but still feel it's maybe a bit off to have some effect happen with no message given

Reply:
I think it'd just be spam honestly, the visual effects exist for a reason